scripts/bug1789259.pl

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env perl
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+use 5.10.1;
+use strict;
+use warnings;
+use lib qw(. lib local/lib/perl5);
+
+use Bugzilla;
+use Bugzilla::Bug;
+use Bugzilla::Constants;
+use Bugzilla::Group;
+use Bugzilla::Search;
+use Getopt::Long;
+use URI qw();
+
+BEGIN { Bugzilla->extensions(); }
+
+# Bug 1789259 - Convert trivial, minor, normal, major+ (for crashes) severities to new style severity
+#
+# severity==trivial or severity==minor -> S4
+# severity==normal -> S3
+# (severity==major or severity==critical or severity==blocker) and (crash signature is not empty or "crash" is in keywords or product==Testing) -> S2
+#
+
+my @products = (
+  "Core",            "Developer Infrastructure",
+  "DevTools",        "External Software Affecting Firefox",
+  "Fenix",           "Firefox",
+  "Firefox for iOS", "Firefox Build System",
+  "GeckoView",       "JSS",
+  "NSPR",            "NSS",
+  "Remote Protocol", "Testing",
+  "Toolkit",         "Web Compatibility",
+  "WebExtensions"
+);
+
+my %severity_queries = (
+  's2' => {
+    product      => \@products,
+    bug_severity => ['major', 'critical', 'blocker'],
+    j_top        => 'OR',
+    f1           => 'cf_crash_signature',
+    o1           => 'isnotempty',
+    f2           => 'keywords',
+    o2           => 'equals',
+    v2           => 'crash',
+    f3           => 'product',
+    o3           => 'equals',
+    v3           => 'Testing',
+  },
```

Comment:
I'm pretty sure this should be limited to open bugs, which wasn't requested in the bug.  I've asked Marco to clarify.

Reply:
Yup - open bugs only.